### DISCUSSION TITLE
Set @disk_controller_mode for additional disks?

### DISCUSSION THREAD

**Discussion by richard-mansfield** [vagrant-libvirt/vagrant-libvirt#1801] [Q&A] [answered]
Setting `disk_bus=scsi` in the domain specifications set the storage controller model to `virtio-scsi` but only for the box image. If I specify more than 8 additional drives via `libvirt.storage :file, :bus="scsi"` the second and subsequent storage controllers revert to `lsilogic`. How can a specify the storage controller model in my case?

```
[root@node1 ~]# lspci -nn | grep "SCSI storage controller"
02:01.0 SCSI storage controller [0100]: Broadcom / LSI 53c895a [1000:0012]
02:02.0 SCSI storage controller [0100]: Broadcom / LSI 53c895a [1000:0012]
03:00.0 SCSI storage controller [0100]: Red Hat, Inc. Virtio SCSI [1af4:1048] (rev 01)
```

The box image I am using does not have support for `LSI 53c895a` although I could inject kmod-sym53c8xx but this would require updating the guest kernel which is not desirable for this project.

**Comment by richard-mansfield** [comment]
FYI, my environment is:
- host O/S is Ubuntu 22.04 (5.15.0-92-generic)
- guess O/S is Rocky Linux (4.18.0-477.27.1)
- vagrant-libvirt (0.12.2, global)
- QEMU emulator version 6.2.0 (Debian 1:6.2+dfsg-2ubuntu6.16)

**Comment by electrofelix** [comment, marked as answer]
Unfortunately it would require reworking the template to apply the disk controller piece to additional disks as well.

https://github.com/vagrant-libvirt/vagrant-libvirt/blob/a94ce0d7b6c90129a38435698ca97b364130313d/lib/vagrant-libvirt/templates/domain.xml.erb#L168

A work around would be to hook a custom class to modify the domain XML after the domain has been defined and before it's started, otherwise it requires patching and a new release

**Comment by richard-mansfield** [comment]
A colleague of mine mentioned a workaround to set `libvirt.storage :file, :bus="sata"` for additional disks.  In which case the 2nd and subsequent storage controllers uses `Intel Corporation 82801IR/IO/IH` 6-port SATA controller.  

```
[root@node1 ~]# lspci -nn | grep "controller" | grep -E "(SCSI|SATA)"
00:1f.2 SATA controller [0106]: Intel Corporation 82801IR/IO/IH (ICH9R/DO/DH) 6 port SATA Controller [AHCI mode] [8086:2922] (rev 02)
02:01.0 SATA controller [0106]: Intel Corporation 82801IR/IO/IH (ICH9R/DO/DH) 6 port SATA Controller [AHCI mode] [8086:2922] (rev 02)
02:02.0 SATA controller [0106]: Intel Corporation 82801IR/IO/IH (ICH9R/DO/DH) 6 port SATA Controller [AHCI mode] [8086:2922] (rev 02)
03:00.0 SCSI storage controller [0100]: Red Hat, Inc. Virtio SCSI [1af4:1048] (rev 01)
```
